### PR TITLE
Let me introduce RowToRowTransformerBase!

### PR DESCRIPTION
**Pull request by Ivanidzo4ka**
I need to have base class from which I can derive OneToOneTransformerBase and some-non OneToOne transforms. We currently have ConcatColumns, TensorFlow and Onnx, and i'm working on NGramHash and StopWordRemover.

### CODE REVIEW

File: src/Microsoft.ML.OnnxTransform/OnnxTransform.cs [hunk 1]
```diff
@@ -111,18 +109,17 @@ private static OnnxTransform Create(IHostEnvironment env, ModelLoadContext ctx)
...
+        private OnnxTransform(IHostEnvironment env, Arguments args, byte[] modelBytes = null) :
+            base(Contracts.CheckRef(env, nameof(env)).Register(nameof(OnnxTransform)))
         {
-            Contracts.CheckValue(env, nameof(env));
-            _host = env.Register(RegistrationName);
-            _host.CheckValue(args, nameof(args));
-            _host.CheckNonWhiteSpace(args.InputColumn, nameof(args.InputColumn));
-            _host.CheckNonWhiteSpace(args.OutputColumn, nameof(args.OutputColumn));
+            Host.CheckValue(args, nameof(args));
+            Host.CheckNonWhiteSpace(args.InputColumn, nameof(args.InputColumn));
```
Comment (sfilipi) [inline src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: >CheckNonWhiteSpace [](start = 17, length = 18)

is this the best check? Would CheckNonNull be better?

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.OnnxTransform/OnnxTransform.cs]: I doubt you can have layer in Onnx which named as " ", also it was code before my change so i hope it's valid :)

File: src/Microsoft.ML.Data/Transforms/RowToRowTransformerBase.cs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                        foreach (var act in disposers)
+                            act();
+                    };
+                }
+                else
+                    disposer = null;
+                return result;
+            }
+
+            protected abstract Delegate MakeGetter(IRow input, int iinfo, Func<int, bool> activeOutput, out Action disposer);
```
Comment (abgoswam) [inline src/Microsoft.ML.Data/Transforms/RowToRowTransformerBase.cs]: >Func<int, bool> activeOutput [](start = 74, length = 28)

follow up question on this -- why add this Func<> delegate to MakeGetter ?

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Transforms/RowToRowTransformerBase.cs]: In this PR mostly for TensorFlow, (but Onnx transform takes same approach)
They have that thing:
activeOutputColNames for caching purpose and only way for me to acquire it is by passing that activeOutput function.
It's not used in other transforms.

Comment (abgoswam) [reply src/Microsoft.ML.Data/Transforms/RowToRowTransformerBase.cs]: thanks for the info. i will look into the TfTransform usage. 

---
In reply to: [235108307](https://github.com/dotnet/machinelearning/pull/1655#discussion_r235108307) [](ancestors = 235108307)

**Pull request merged**